src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6973,7 +6990,20 @@ public override IOperation VisitSwitchExpression(ISwitchExpressionOperation oper
             SpillEvalStack();
             RegionBuilder resultCaptureRegion = CurrentRegionRequired;
             int captureOutput = captureIdForResult ?? GetNextCaptureId(resultCaptureRegion);
-            var capturedInput = VisitAndCapture(operation.Value);
+            IOperation? capturedInput;
+
+            // There is only one discard pattern operation. We evaluate the input always, but it won't
+            // be used in an arm pattern so we don't capture
+            if (operation.Arms.Length == 1 && operation.Arms[0].Pattern is IDiscardPatternOperation)
+            {
+                capturedInput = null;
+                Visit(operation.Value);
```

Comment:
>Visit(operation.Value); [](http://example.com/codeflow?start=16&length=23)

I think this drops the result of the visit on the floor. Should we add the result as a statement?